### PR TITLE
(WIP) Add vpc peering

### PR DESCRIPTION
**Pull request by sevenmachines**
This change allows us to set up a config with,

```
  vpc:
      peering:
        helloworld-dev1: {}
```

and run `fab enable_vpc_peering' to automatically create a peering connection with that stack name match. Currently the both vpc's overall CIDR will be routed between all route tables. The empty dictionary allows for expansion of configuration settings into more granular control. 

```
* Create new class VPC to handle vpc connections and peering
* Update requirements to include boto3
* Add new fab tasks enable/disable_vpc_peering
* Add boto3 to setup.py requirements
* Add VPC docs page
* Add vpc requester/accepter routes to all route tables in each vpc
```

### CODE REVIEW

**Comment by sevenmachines** [comment]
needs some function docs

File: docs/vpc-configuration.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# VPC Configuration
+
+### Contents
+
+##### [1. Basic configuration](#basic-configuration)
+
+VPC configuration is done in the under the 'vpc' key section of the cloudformation configuration file.
+
+Defaults are set so that without specifying anything, we get a VPC with CIDR `10.128.0.0/16` and 3 subnets, `10.0.0.0/20`, `10.0.16.0/20`, and `10.0.32.0/20`
```
Comment (mattpep) [inline docs/vpc-configuration.md:9]: The 3 /20 subnets do not fall within the /16 . (if there's a reason for this then it's not clear to me)

File: docs/vpc-configuration.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+* [Peered VPCs](#peered-vpcs)
+
+### Basic configuration 
+
+
+### Peering
+
+Using the VPC class, we can setup up peering to another VPC by calling enable_peering. This call will then
+
+* Use the cloudformation configuration file to get a list of stacks to peer to
```
Comment (mattpep) [inline docs/vpc-configuration.md:31]: Does this mean that one can only peer to stacks defined in the same .yml file ? (and not ones from other repos)

**Comment by sevenmachines** [comment]
Hold on on merging this, I'm going to simplify some of the method signatures for the more advanced configuration version. Maybe make it just the one PR

File: docs/vpc-configuration.md (hunk 2 above)
Comment (sevenmachines) [inline docs/vpc-configuration.md:31]: no, we give a stack name search term, and get a 1:1 mapping to a real stack_name in the same account. Being in the same account is the only restriction so far

File: docs/vpc-configuration.md (hunk 1 above)
Comment (sevenmachines) [inline docs/vpc-configuration.md:9]: cocked up documentation, my apologies

**Comment by sevenmachines** [comment]
Deprecating this in favour of https://github.com/ministryofjustice/bootstrap-cfn/pull/166

**Pull request closed without merging**